### ISSUE TITLE
About Saving results

### ISSUE THREAD

**Issue by sudarshan-s-harithas** [ManipTrans/ManipTrans#8]
Hi, 

This is a cool work; thanks for releasing the code. 

I'm working on saving the wrist trajectory, joint angles, and the corresponding object states of the Inspire hands, after the RL based retargetting is complete. Could you suggest the best way to save this data?

Thank You

**Comment by KailinLi** [comment]
Thank you for your interest in our ManipTrans work!

We have implemented code to save the simulation states. After you have trained the Residual Policy network, you can try the following command:

```bash
python main/rl/train.py task=ResDexHand dexhand=inspire side=BiH headless=true num_envs=512 learning_rate=2e-4 test=true randomStateInit=false dataIndices=[20aed@0] rh_base_model_checkpoint=assets/imitator_rh_inspire.pth lh_base_model_checkpoint=assets/imitator_lh_inspire.pth actionsMovingAverage=0.4 checkpoint=runs/cross_20aed@0_inspire__xxxxxx/nn/cross_20aed@0_inspire.pth save_rollouts=true num_rollouts_to_save=32 num_rollouts_to_run=8192 save_successful_rollouts_only=true
```

In this command:
- Setting `headless=true` speeds up the dumping process.
- `num_rollouts_to_run=8192` indicates that 8192 rollouts will be attempted.
- `num_rollouts_to_save=32` specifies that 32 rollout trajectories will be saved into HDF5 file (if there are 32 successful rollouts).

The resulting trajectories will be saved in the `dumps` folder. You can read the HDF5 files and check whether the saved values meet your requirements.

If you have any further questions, please feel free to ask us.

**Comment by sudarshan-s-harithas** [comment]
Thanks a lot for the detailed response. I have a follow-up question. When looking into the successful runs, I find the following fields "actions, dq_lh, dq_rh, joint_state_lh, joint_state_rh, q_lh, q_rh, reward, state_lh, state_manip_obj_lh, state_manip_obj_rh, state_rh, tip_force_lh, tip_force_rh"

**Comment by KailinLi** [comment]
You're most welcome!

Taking *the right hand* as an example, the joint angles of the fingers can be found in `q_rh`, which has dimensions `[T, 12]`, corresponding to the 12 degrees of freedom of the Inspire hand. Additionally, `dq_rh` represents the velocities of these 12 DoFs.

For the SE(3) pose of the wrist, you'll need to parse the variable `state_rh`, which has dimensions `[T, 13]`. This variable contains:

- 3-dimensional *translation*,
- 4-dimensional *rotation parameters in quaternion* form `(x, y, z, w)`,
- 3-dimensional linear velocity,
- 3-dimensional angular velocity.

You'll need to manually assemble these components to construct the SE(3) pose. For more detailed information, you might want to refer to the official documentation of Isaac Gym.

If there's anything else I can help with, please feel free to ask!

**Comment by sudarshan-s-harithas** [comment]
Thanks a lot for the detailed explanation. 

I am also looking forward to your dataset release; please let me know if you have any expected time by which you plan to release the data. 

Congrats again on the great work.

**Comment by KailinLi** [comment]
Thank you very much for your interest!

In fact, we have already prepared all the data (exported in the HDF5 format), as well as a dataloader that allows for one-click visualization.

We are currently performing the final checks, and we hope to organize all the documentation and upload everything to Hugging Face within the next five days.

**Comment by huamo555** [comment]
> Hi,
> 
> This is a cool work; thanks for releasing the code.
> 
> I'm working on saving the wrist trajectory, joint angles, and the corresponding object states of the Inspire hands, after the RL based retargetting is complete. Could you suggest the best way to save this data?
> 
> Thank You

Hello, have you successfully reproduced the metrics of the paper? Do you know in which part of the code the metrics for calculating the success rate in the paper are located? Thank You！
